woocommerce/woocommerce-android
# Use summary visitor stats

Closes: #10946

### Why
Using the summary stats endpoint, it is more accurate to calculate visitor stats than the current behavior of summing the site stats interval.
more context here: pe5pgL-1p6-p2

### Description
This PR updates how we fetch visitor stats in the Analytic Hub screen, favoring the usage of the summary stats endpoint instead of using the site stats endpoint and summing the interval values. The new summary endpoint lacks support for quarter granularities, and custom ranges bigger than 1 day. For those two special cases, we will:
- In the case of quarter granularities, rely on the previous behavior of summing the site stats interval
- For custom ranges bigger than 1 day, display a message indicating that the visitor stats for this range are not available.

#### Design
<img width="300" src="https://github.com/woocommerce/woocommerce-android/assets/18119390/7154c3b1-21aa-4502-8eed-d0ce4e8aa489" />


### Testing instructions
TC1
1. Open the app
2. Tap on See all store analytics
3. Check that the session stats are displayed and accurate

TC2
1. Open the app
2. Tap on See all store analytics
3. Tap on the range selection control
4. Select the Custom option
5. Select a range of dates bigger than 1 day
6. Check that the session card displays a not available message

### Images/gif

https://github.com/woocommerce/woocommerce-android/assets/18119390/844ad085-3480-4519-9a00-8168a5159d83
